app/src/Flashcards/DictionarySearch.js

```diff
@@ -1,6 +1,30 @@
 import * as React from "react";
 
-const DictionarySearch = () => {
-  return <input placeholder="search"></input>;
+import useApi from "../auth/useApi";
+
+const DictionarySearch = ({ front, setFront, back, setBack }) => {
+  const { apiClient } = useApi();
+  const getDefinition = async (event) => {
+    event.preventDefault();
+    const result = await apiClient.getDefinition(front);
+    setBack(result);
+  };
+  return (
+    <>
+      <form onSubmit={getDefinition}>
+        <label htmlFor="definition">Check out dictionary definitions</label>
+        <input
+          id="definition"
+          name="defintitionSearch"
+          type="text"
+          value={front}
+          onChange={(e) => setFront(e.target.value)}
+          placeholder="search"
+          required
+        ></input>
+        <input type="submit" value="search"></input>
+      </form>
+    </>
+  );
```

Comment:
So the actions of fetching the definition from the API and setting the definition as the back of the card are combined into one action?  If the user doesn't like the definition that was returned, do they need to click back to "Edit Card" so they can specify their own definition?

Reply:
no they don't need to click on "edit", they can simply edit the defintion right away in the form and then save it.